### ISSUE TITLE
After building with cmake, how do you load the build onto an emulator?

### ISSUE THREAD

**Issue by Takkobay** [elucideye/drishti#736]
I managed to successfully build using cmake, but I am having trouble figuring out how to load the code onto an emulator through android studio.

**Comment by headupinclouds** [comment]
I'm not sure what you mean by "the code".  If you are referring to the real-time FaceFilter app, I don't think it will be easy to run that in an emulator.  You would have to create some kind of camera interface to support testing.   There are unit tests that run the drishti library using the Android QEMU emulator.  If you are looking for the latter, I can provide some links.

**Comment by Takkobay** [comment]
Yes, the links will be helpful. Also, sorry for not clarifying, but I did mean the real time FaceFilter App. Also for the camera interface you mentioned, do you mean that I would have to link the emulator to a webcam? or do you mean that I would have to develop some sort of UI to support a camera?

**Comment by headupinclouds** [comment]
> or do you mean that I would have to develop some sort of UI to support a camera?

AFAIK, the Android emulator doesn't come with camera support.  I did check at one point to see if end-to-end application level testing would be possible.   I think you would have to write some code to play back video files or something like that.

Why do you need it?

If you manage your unit tests with gauze, it manage the emulator for you:

https://github.com/hunter-packages/gauze

This  allows testing OpenGL ES shader pipeline in Travis CI.

See the  docs (in particular the Amdroid section):

https://gauze.readthedocs.io/en/latest/prereq/android.html

Example (`gauze_main()`)

https://github.com/elucideye/drishti/blob/378725771b0b4353e5b3fa93f32175023dfc0e0f/src/tests/drishti/test-drishti.cpp#L38

Or any of the tests in the gauze repo:

https://github.com/hunter-packages/gauze/tree/master/test/gauze

**Comment by Takkobay** [comment]
Thank you, I'll take a look at the documentation. I'm running it on an emulator because I don't have an android phone handy. As for why I'm interested in running the example code, I want to see how well the library runs on mobile to see if I can apply the library to a research project I'm a part of. My plan would be for the code to run in the background to calculate distance between the phone and face and to also possibly calculate gaze. The calculated information would then be used for image processing variables.
